### PR TITLE
Allows to add rows into sheets already created

### PR DESCRIPTION
**Pull request by idlira**
A new method allow us to select the Sheet to add rows, needed when we need to jump back and forth across sheets.

Note that this won't change the actual visible sheet once Excel opens. The first sheet remains as the visible one. A method could be added to change the current visible sheet if the need arrises.

Fixes: OX-6630

### CODE REVIEW

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/web/data/ExcelExport.java [hunk 1]
```diff
@@ -225,6 +228,21 @@ public void createSheet(@Nullable String name) {
         currentSheet.setRepeatingRows(new CellRangeAddress(0, 0, -1, -1));
     }
 
+    /**
+     * Sets the sheet provided as the current sheet to receive data.
+     * <p>
+     * Note that this won't change the Sheet visible when the file is open.
```
Comment (bwiedmann) [inline src/main/java/sirius/web/data/ExcelExport.java:234]:
```suggestion
     * Note that this won't set the Sheet active when the file is open.
```

**Review by bwiedmann — APPROVED** [review]

File: src/main/java/sirius/web/data/ExcelExport.java [hunk 2]
```diff
@@ -65,8 +67,8 @@
 
     private final Workbook workbook;
     private Sheet currentSheet;
-    private int rows = 0;
-    private int maxCols = 0;
+    private final Map<String, Integer> rows = new HashMap<>();
```
Comment (andyHa) [inline src/main/java/sirius/web/data/ExcelExport.java:70]: AtomicInteger? this has incrementAndGet bultin....

File: src/main/java/sirius/web/data/ExcelExport.java [hunk 3]
```diff
@@ -509,8 +527,10 @@ public void writeToStream(OutputStream stream) {
     }
 
     private void addAutoFilter() {
```
Comment (andyHa) [inline src/main/java/sirius/web/data/ExcelExport.java:529]: this should be applied to all sheets?!

File: src/main/java/sirius/web/data/ExcelExport.java [hunk 4]
```diff
@@ -519,7 +539,8 @@ private void autosizeColumns() {
             // we do not want to autosize columns of streamed excel sheets, because of performance reasons
```
Comment (andyHa) [inline src/main/java/sirius/web/data/ExcelExport.java:539]: this should be applied to all sheets?!

File: src/main/java/sirius/web/data/ExcelExport.java [hunk 5]
```diff
@@ -583,4 +604,11 @@ public ExcelExport withMaxRowsReachedHandler(Consumer<String> maxRowsReachedHand
         this.maxRowsReachedHandler = maxRowsReachedHandler;
         return this;
     }
+
+    private int getAndIncrementRowForSheet(String name) {
```
Comment (andyHa) [inline src/main/java/sirius/web/data/ExcelExport.java:608]: AtomicInteger#incrementAndGet

**Review by andyHa — CHANGES_REQUESTED** [review]

**Review by bwiedmann — APPROVED** [review]

File: src/main/java/sirius/web/data/ExcelExport.java [hunk 6]
```diff
@@ -65,8 +68,8 @@
 
     private final Workbook workbook;
     private Sheet currentSheet;
-    private int rows = 0;
-    private int maxCols = 0;
+    private final Map<String, AtomicInteger> rows = new HashMap<>();
```
Comment (andyHa) [inline src/main/java/sirius/web/data/ExcelExport.java:71]: shoudln't we use the sheet index here, as the name can be non-unique(?)

Comment (idlira) [reply src/main/java/sirius/web/data/ExcelExport.java]: Names are unique, but pushing a version based on the index.

**Review by bwiedmann — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Pull request merged**